internal/auth/oidc/repository_auth_method_operational_state.go

```diff
@@ -0,0 +1,100 @@
+package oidc
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/hashicorp/boundary/internal/db"
+	"github.com/hashicorp/boundary/internal/errors"
+	"github.com/hashicorp/boundary/internal/kms"
+	"github.com/hashicorp/boundary/internal/oplog"
+)
+
+// MakeInactive will transision an OIDC auth method from either the
+// ActivePrivateState or the ActivePublicState to the InactiveState.
+// No options are supported.
+func (r *Repository) MakeInactive(ctx context.Context, authMethodId string, version uint32, _ ...Option) (*AuthMethod, error) {
+	const op = "oidc.(Repository).MakeInactive"
+	updated, err := r.transitionAuthMethodTo(ctx, authMethodId, InactiveState, version)
+	if err != nil {
+		return nil, errors.Wrap(err, op)
+	}
+	return updated, nil
+}
+
+// MakePrivate will transision an OIDC auth method from either the
+// InactiveState or the ActivePublicState to the ActivePrivateState.  If
+// transitioning from the InactiveState, the transition will only succeed if
+// the oidc.ValidateAuthMethod(...) succeeds. No options are supported.
+func (r *Repository) MakePrivate(ctx context.Context, authMethodId string, version uint32, _ ...Option) (*AuthMethod, error) {
+	const op = "oidc.(Repository).MakePrivate"
+	updated, err := r.transitionAuthMethodTo(ctx, authMethodId, ActivePrivateState, version)
+	if err != nil {
+		return nil, errors.Wrap(err, op)
+	}
+	return updated, nil
+}
+
+// MakePublic will transision an OIDC auth method from either the
+// InactiveState or the ActivePrivateState to the ActivePublicState.  If
+// transitioning from the InactiveState, the transition will only succeed if
+// the oidc.ValidateAuthMethod(...) succeeds. No options are supported.
+func (r *Repository) MakePublic(ctx context.Context, authMethodId string, version uint32, _ ...Option) (*AuthMethod, error) {
+	const op = "oidc.(Repository).MakePublic"
+	updated, err := r.transitionAuthMethodTo(ctx, authMethodId, ActivePublicState, version)
+	if err != nil {
+		return nil, errors.Wrap(err, op)
+	}
+	return updated, nil
+}
+
+func (r *Repository) transitionAuthMethodTo(ctx context.Context, authMethodId string, desiredState AuthMethodState, version uint32, _ ...Option) (*AuthMethod, error) {
+	const op = "oidc.(Repository).updateOperationalState"
```

Comment:
The op const does not match the method name.

Reply:
ty!